students/calvin_fannin/Lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python3
+
+
+"""
+Class to hold a information for one donor first name lastname and donation amount
+"""
+class Donor():
+    def __init__(self, firstname, lastname,donation):
+        self.fname = firstname.title().strip()
+        self.lname = lastname.title().strip()
+        if donation <= 0:
+            donationError = ValueError('Donation cannot be negative')
+            raise donationError
+        else:
+            self.donations = [donation]
+        self.dc = DonorCollection()
+
+    @property
+    def fullname(self):
+       """ Returns the first and lastname as one string """
+       return self.fname + " " + self.lname
+
+    @property
+    def totaldonation(self):
+        """ Returns the sum of all donations for a donor """
+        return sum(self.donations)
+
+    def add_donation(self,donation):
+        """ Returns the sum of all donations for a donor """
+        if donation < 0:
+            donationError = ValueError('Donation cannot be negative')
+            raise donationError
```

Comment:
a custom exception is a class so it will follow a class naming convention (CamelCase) -> `raise DonationError`